### PR TITLE
Enable V2 by default

### PR DESCRIPTION
**Pull request by davidor**
### What does this PR do?

Enables V2 by default. It's no longer necessary to run the commands with `KUSTOMIZE_CONFIG=config/test-v2`.
I've added some configs to be able to test V1 in case we need it.

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]
No review necessary from docs on this one

File: config/test-v1/manager_enablev1_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# The following patches adds configure the DatadogAgents CRD to use v1alpha1 as the storage version.
```
Comment (celenechang) [inline config/test-v1/manager_enablev1_patch.yaml:1]:
```suggestion
# The following patches adds a configuration to the DatadogAgents CRD to use v1alpha1 as the storage version.
```

File: config/test-v1/storagev1_in_datadogagents.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
-# The following patches adds configure the DatadogAgents CRD to use v2alpha1 as the storage version.
+# The following patches adds configure the DatadogAgents CRD to use v1alpha1 as the storage version.
```
Comment (celenechang) [inline config/test-v1/storagev1_in_datadogagents.yaml:1]:
```suggestion
# The following patches adds a configuration to the DatadogAgents CRD to use v1alpha1 as the storage version.
```

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 3]
```diff
@@ -921,7 +921,7 @@ type DatadogAgentStatus struct {
 // DatadogAgent Deployment with the Datadog Operator.
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
-// +kubebuilder:unservedversion
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:924]: should we add this flag to v1alpha1?

File: docs/how-to-contribute.md [hunk 4]
```diff
@@ -32,18 +32,16 @@ make IMG=test/operator:test deploy
 
 Note: `IMG` currently defaults to: `datadog/datadog-operator:latest`
 
-## \[TMP\] how to test `v2alpha`
+## Test using `v2alpha`
```
Comment (celenechang) [inline docs/how-to-contribute.md:35]: should we rm this section header and merge the cert manager install cmd with the section above?

Comment (davidor) [reply docs/how-to-contribute.md]: Changed :+1:

File: config/test-v1/manager_enablev1_patch.yaml (hunk 1 above)
Comment (davidor) [reply config/test-v1/manager_enablev1_patch.yaml]: Fixed.

File: config/test-v1/storagev1_in_datadogagents.yaml (hunk 2 above)
Comment (davidor) [reply config/test-v1/storagev1_in_datadogagents.yaml]: Fixed.

File: apis/datadoghq/v2alpha1/datadogagent_types.go (hunk 3 above)
Comment (davidor) [reply apis/datadoghq/v2alpha1/datadogagent_types.go:924]: Added and adapted manifests, patches, etc. :+1:

**Review by CharlyF — APPROVED** [review]
Tested changes, all set.

**Pull request merged**